sonar/modules/documents/api.py

```diff
@@ -138,6 +143,118 @@ def get_record_by_identifier(cls, identifiers):
+        """
+        # For documents, a thumbnail and a fulltext file is generated.
+        if not deleted:
+            self.create_fulltext_file(self.files[file.key])
+            self.create_thumbnail(self.files[file.key])
+
+        super(DocumentRecord, self).sync_files(file, deleted)
+
+    def create_fulltext_file(self, file):
+        """Create fulltext file corresponding to give file object.
+
+        :param file: File object.
+        """
+        # If extract fulltext is disabled or file is not a PDF
+        if not current_app.config.get(
+                'SONAR_DOCUMENTS_EXTRACT_FULLTEXT_ON_IMPORT'
+        ) or file.mimetype != 'application/pdf':
+            return
+
+        # Try to extract full text from file data, and generate a warning if
+        # it's not possible. For several cases, file is locked against fulltext
+        # copy.
+        try:
+            with file.file.storage().open() as pdf_file:
+                fulltext = extract_text_from_content(pdf_file.read())
+
+            key = change_filename_extension(file.key, 'txt')
+            self.files[key] = BytesIO(fulltext.encode())
+            self.files[key]['type'] = 'fulltext'
+        except Exception as exception:
+            current_app.logger.warning(
+                'Error during fulltext extraction of {file} of record '
+                '{record}: {error}'.format(file=file.key,
+                                           error=exception,
+                                           record=self['identifiedBy']))
+
+    def create_thumbnail(self, file):
+        """Create a thumbnail for record.
+
+        This is done by getting the file with order 1 or the first file
+        instead.
+
+        :param file: File from which thumbnail is created.
+        """
+        try:
+            # Create thumbnail
+            image_blob = create_thumbnail_from_file(file.file.uri,
+                                                    file.mimetype)
+
+            thumbnail_key = change_filename_extension(file['key'], 'jpg')
+
+            # Store thumbnail in record's files
+            self.files[thumbnail_key] = BytesIO(image_blob)
+            self.files[thumbnail_key]['type'] = 'thumbnail'
+        except Exception as exception:
+            current_app.logger.warning(
+                'Error during thumbnail generation of {file} of record '
+                '{record}: {error}'.format(file=file['key'],
+                                           error=exception,
+                                           record=self.get(
+                                               'identifiedBy', self['pid'])))
+
+    def get_main_file(self):
+        """Get the main file of record."""
+        files = [file for file in self.files if file.get('type') == 'file']
+
+        if not files:
+            return None
+
+        for file in files:
+            if file.get('type') == 'file' and file.get('order') == 1:
+                return file
```

Comment:
You don't need to check again `file.get('type') == 'file'` here because you already filters file list at line 247.
So you rewrite simply : 
```
for file in files if file.get('order') == 1: 
    return file
```

Reply:
Done